pkg/integrations/v2/register.go

```diff
@@ -86,9 +86,12 @@ type UpgradedConfig interface {
 type Type int
 
 const (
+	// TypeInvalid is an invalid type.
+	TypeInvalid Type = iota
```

Comment:
Introducing TypeInvalid as the zero value removes the need for examining `ok` from RegisteredType